resources/azure/compute/vm.go

```diff
@@ -74,23 +86,60 @@ func NewWindowsInstance(e azure.Environment, name, imageUrn, instanceType string
 		}
 	}
 
-	vm, nw, err := newVMInstance(e, name, imageUrn, instanceType, windowsOsProfile)
+	vm, nw, err := newVMInstance(e, name, imageUrn, instanceType, windowsOsProfile, opts...)
 	if err != nil {
-		return nil, nil, pulumi.StringOutput{}, err
+		return nil, pulumi.StringOutput{}, pulumi.StringOutput{}, err
 	}
 
-	return vm, nw, windowsAdminPassword.Result, nil
+	publicKey, err := os.ReadFile(e.DefaultPublicKeyPath())
+	if err != nil {
+		return nil, pulumi.StringOutput{}, pulumi.StringOutput{}, err
+	}
+
+	setupSSHCommand, err := compute.NewVirtualMachineRunCommandByVirtualMachine(e.Ctx(), fmt.Sprintf("%s-init-cmd", name), &compute.VirtualMachineRunCommandByVirtualMachineArgs{
+		ResourceGroupName: pulumi.String(e.DefaultResourceGroup()),
+		VmName:            vm.Name,
+		AsyncExecution:    pulumi.Bool(false),
+		RunCommandName:    pulumi.String("InitVM"),
+		Source: compute.VirtualMachineRunCommandScriptSourceArgs{
+			Script: pulumi.String(setupSSHScriptContent),
+		},
+		Parameters: compute.RunCommandInputParameterArray{
+			compute.RunCommandInputParameterArgs{
+				Name:  pulumi.String("authorizedKey"),
+				Value: pulumi.String(publicKey),
+			},
+		},
+		TimeoutInSeconds:                pulumi.Int(120),
+		TreatFailureAsDeploymentFailure: pulumi.Bool(true),
+	}, pulumi.Parent(vm))
+
+	if err != nil {
+		return nil, pulumi.StringOutput{}, pulumi.StringOutput{}, err
+	}
+
+	privateIP = pulumi.All(nw.IpConfigurations.Index(pulumi.Int(0)).PrivateIPAddress().Elem(), setupSSHCommand.URN()).ApplyT(func(args []interface{}) string {
+		return args[0].(string)
+	}).(pulumi.StringOutput)
+
+	return vm, privateIP, windowsAdminPassword.Result, nil
 }
 
-func newVMInstance(e azure.Environment, name, imageUrn, instanceType string, osProfile compute.OSProfilePtrInput) (*compute.VirtualMachine, *network.NetworkInterface, error) {
+func newVMInstance(e azure.Environment, name, imageUrn, instanceType string, osProfile compute.OSProfilePtrInput, opts ...pulumi.ResourceOption) (*compute.VirtualMachine, *network.NetworkInterface, error) {
```

Comment:
Note for reviewers: this allows passing the parent to all the resources, so that they belong to the parent component, similar to what we have in `aws`